### PR TITLE
Course Review

### PR DESCRIPTION
**Pull request by megatimranikram**

### CODE REVIEW

**Comment by megatimranikram** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+class CoursesController < ApplicationController
+  module UsersController
```
Comment (manhnt-0960) [inline app/controllers/courses_controller.rb]: what do this line mean ?

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  def index
+    @course = Course.page.order("created_at DESC")
+  end
+
+  def show
+    @course = Course.find(params[:id])
```
Comment (manhnt-0960) [inline app/controllers/courses_controller.rb]: let's use find_by instead of find

File: app/views/courses/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<div class="col-md-6 col-md-offset-3">
+
+	<div class="new-course-form">
```
Comment (manhnt-0960) [inline app/views/courses/new.html.erb]: let's check indent

File: config/routes.rb [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
   get 'password_resets/new'
   get 'password_resets/edit'
+  resources :courses
+  root 'courses#index'
```
Comment (manhnt-0960) [inline config/routes.rb]: delete this line, because it not use

**Comment by megatimranikram** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    def course_params
+      params.require(:course).permit(:title, :description, :credits)
+    end
+
+    def find_course
+      @course = Course.find_by :id params[:id]
+    end
+  end
+end
```
Comment (manhnt-0960) [inline app/controllers/courses_controller.rb]: remove `end` keyword because of redundant

File: app/controllers/courses_controller.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    @course = Course.new(course_params)
+
+    if @course.save
+      redirect_to root_path
+    else
+      render :new
+    end
+  end
+
+  def edit
```
Comment (manhnt-0960) [inline app/controllers/courses_controller.rb]:
```suggestion
  def edit; end
```

File: app/controllers/courses_controller.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  private
+
+    def course_params
+      params.require(:course).permit(:title, :description, :credits)
+    end
+
+    def find_course
+      @course = Course.find_by :id params[:id]
```
Comment (manhnt-0960) [inline app/controllers/courses_controller.rb:47]: handle case return nil

File: app/helpers/courses_helper.rb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+module CoursesHelper
```
Comment (manhnt-0960) [inline app/helpers/courses_helper.rb]: delete this file

File: app/views/courses/_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<%= creditsform_for @course do |f| %>
+	<%= select_tag(:category_id, options_for_select(@categories), :prompt => "Select a category") %>
```
Comment (manhnt-0960) [inline app/views/courses/_form.html.erb]: indent 2 spaces

File: app/views/courses/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,6 @@
+<div class="col-md-6 col-md-offset-3">
+  <div class="new-course-form">
+		<h1>New Course</h1>
```
Comment (manhnt-0960) [inline app/views/courses/new.html.erb:3]: indent

**Comment by megatimranikram** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    redirect_to_root_path
+  end
+
+  private
+
+    def course_params
+      params.require(:course).permit(:title, :description, :credits)
+    end
+
+    def find_course
```
Comment (ngocvt-0484) [inline app/controllers/courses_controller.rb:46]: Where is this method used?

File: app/controllers/courses_controller.rb [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+      @course = Course.find_by :id params[:id]
+      if @course.find_course course_params
```
Comment (ngocvt-0484) [inline app/controllers/courses_controller.rb:48]: 48 -> 54 wrong, dont need it

File: app/views/courses/_form.html.erb [hunk 13]
```diff
@@ -0,0 +1,8 @@
+<%= creditsform_for @course do |f| %>
```
Comment (ngocvt-0484) [inline app/views/courses/_form.html.erb:1]: form_for ?

**Pull request closed without merging**